src/ProxyClient/Varnish.php

```diff
@@ -35,8 +35,9 @@ public function __construct(
 
     private function fetchAndMergeAuthHeaders(array $headers): array
     {
-        if ($this->configResolver->hasParameter('http_cache.varnish_invalidate_token')) {
-            $headers[InvalidateTokenController::TOKEN_HEADER_NAME] = $this->configResolver->getParameter('http_cache.varnish_invalidate_token');
+        if ($this->configResolver->hasParameter('http_cache.varnish_invalidate_token') &&
+            '' !== $token = $this->configResolver->getParameter('http_cache.varnish_invalidate_token')) {
```

Comment:
CS: this is not readable to me
```suggestion
        if (
            $this->configResolver->hasParameter('http_cache.varnish_invalidate_token')
            && '' !== ($token = $this->configResolver->getParameter(
                'http_cache.varnish_invalidate_token'
            ))
        ) {
```